### PR TITLE
Updatedocs

### PR DESCRIPTION
**Pull request by DanielMorsing**
This PR moves the documentation to read-the-docs based jekyll site and adds the migration plan for v0.7 changes

### CODE REVIEW

File: docs/5_migrate.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## ClusterGitTrack migration plan
+Kubernetes disallows having owner relationships that go from one namespace into another or from a namespaced object to a cluster-scoped one. Previously, Faros has been able to create these, and because of limitations in the Kubernetes garbage collector, it hasn’t been immediately obvious that this shouldn’t have worked.
+
+This document sets out how to check whether you might be vulnerable to this issue and how to mitigate it when upgrading to Faros version 0.7.0
+
+## Checking if you’re impacted
+
+If you have any ClusterGitTrackObjects in your cluster, you are impacted. You can check this by running `kubectl get clustergittrackobjects`
+
+If you don’t have any ClusterGitTrackObjects, you might still be impacted. Run the tool [here](https://github.com/pusher/faros/blob/master/hack/namespacecheck/namespacechecker.go) to check if you have any GitTrackObjects owned by GitTracks in a different namespace
```
[the remark this answers is not in the document]
Comment (DanielMorsing) [reply docs/5_migrate.md]: The file includes a comment on how to run it

File: docs/5_migrate.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+The rest of this document sets out how to migrate in the case that you are impacted.
+
+## Migrating ClusterGitTrackObjects
+
+If you have ClusterGitTrackObjects in your setup, then you will have to migrate those to being managed by ClusterGitTracks
+
+
+1. Scale down your Faros deployments so that there are no active Faros pods running
+2. Remove all `ownerReferences` from ClusterGitTrackObjects
```
[the remark this answers is not in the document]
Comment (DanielMorsing) [reply docs/5_migrate.md]:
```
kubectl get -a clustergittrackobjects -o json > /tmp/cgtos.json
jq '.items[].metadata.ownerReferences = null' /tmp/cgtos.json > /tmp/newcgtos.json
kubectl apply -f /tmp/newcgtos.json
```

It's a bit hacky. Still want it in the docs?

File: docs/5_migrate.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+3. For every `GitTrack` which previously owned a `ClusterGitTrackObject`, create a `ClusterGitTrack` that matches its target.
```
[the remark this answers is not in the document]
Comment (DanielMorsing) [reply docs/5_migrate.md]: The tool should output a file that you can just apply and you should be good. I'll add some text about this here

File: docs/5_migrate.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+4. Create a new deployment of Faros with the flags `--gittrack-mode=Disabled` and `--clustergittrack-mode=ExcludeNamespaced`
+5. Check that ClusterGitTrackObjects are now owned by ClusterGitTracks
```
[the remark this answers is not in the document]
Comment (DanielMorsing) [reply docs/5_migrate.md]: This was done manually when I did it on kari

File: hack/namespacecheck/namespacechecker.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	}
+
+	uidToNs := make(map[string]string)
+	for _, i := range gtjson.Items {
+		uidToNs[string(i.GetUID())] = i.GetNamespace()
+	}
+
+	var foundOne bool
+	for _, i := range gtojson.Items {
+		for _, owner := range i.GetOwnerReferences() {
```
[the remark this answers is not in the document]
Comment (DanielMorsing) [reply hack/namespacecheck/namespacechecker.go]: It should give an empty `ownerNs` and never match on what came out of `GetNamespace()`

File: hack/namespacecheck/namespacechecker.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+func makeCGTs(gtjson faros.GitTrackList, cgtofile []byte) []byte {
+	// parse the cgtfile to figure out which GitTracks owns it
+	var cgtojson faros.ClusterGitTrackObjectList
+	err := json.Unmarshal(cgtofile, &cgtojson)
+	if err != nil {
+		panic(err)
+	}
+
+	sourceUIDSet := make(map[string]bool)
```
[the remark this answers is not in the document]
Comment (JoelSpeed) [reply hack/namespacecheck/namespacechecker.go]: Can I get an ack that you've seen this comment above?

File: hack/namespacecheck/namespacechecker.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+		}
+	}
+	var toCreate []faros.GitTrack
+	for _, i := range gtjson.Items {
+		if sourceUIDSet[string(i.GetUID())] {
+			toCreate = append(toCreate, i)
+		}
+	}
+
+	var cgtjson faros.ClusterGitTrackList
```
[the remark this answers is not in the document]
Comment (JoelSpeed) [reply hack/namespacecheck/namespacechecker.go]: Bump

File: docs/4_advanced.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+```
+kubectl delete gittrack --cascade=false <gittrack-name>
+```
+
+If you wish to remove Faros entirely, we recommend deleting all `GitTrack` and
+then `GitTrackObject` and `ClusterGitTrackObject` resources using the
+`--cascade=false` flag.
+
+### Three Way Merge
```
Comment (JoelSpeed) [inline docs/4_advanced.md]: Just noticed this is an H3 vs the rest at H2, we should probably fix that

**Pull request merged**